pkg/cmd/roachprod/main.go

```diff
@@ -1171,6 +1172,45 @@ func validateAndConfigure(cmd *cobra.Command, args []string) {
 	}
 }
 
+var updateCmd = &cobra.Command{
+	Use:   "update",
+	Short: "check TeamCity for a new roachprod binary and update if available",
+	Long: "Will attempt to download the latest master branch roachprod binary from teamcity" +
+		" and swap the current roachprod with it. The current roachprod binary will be backed up" +
+		" and can be restored via `roachprod update --revert`.",
+	Run: wrap(func(cmd *cobra.Command, args []string) error {
+		currentBinary, err := os.Executable()
+		if err != nil {
+			return err
+		}
+
+		if revertUpdate {
```

Comment:
Do we have a concrete use case for this feature? To me it would be something that we add only if we know it's needed frequently. The main things I'm not a big fan of are:

1) it may be hard to know what the "previous" version is (what we are reverting to), especially in the case where the previous binary was built with `dev` (i.e., without stamping).
2) For folks that have roachprod in their `$PATH`, this will add `roachprod.bak` to the same directory, which can be confusing and/or inconvenient when doing `roachp<tab>`.